src/CoolPropLib.cpp

```diff
@@ -18,27 +18,27 @@
 
 #include <string.h>
 
-bool str2buf(const std::string& str, char * buf, int n)
+void str2buf(const std::string& str, char * buf, int n)
 {
-  if (str.size() < static_cast<unsigned int>(n)) {
+  if (str.size() < static_cast<unsigned int>(n))
     strcpy(buf, str.c_str());
-    return true;
-  }
-  return false;
+  else
+    throw CoolProp::ValueError("Buffer size is too small");
 }
 
 // In Microsoft Excel, they seem to check the FPU exception bits and error out because of it.  
 // By calling the _clearfp(), we can reset these bits, and not get the error
 // See also http://stackoverflow.com/questions/11685441/floating-point-error-when-calling-dll-function-from-vba/27336496#27336496
 // See also http://stackoverflow.com/questions/16849009/in-linux-do-there-exist-functions-similar-to-clearfp-and-statusfp for linux and OSX
-void reset_fpu()
-{
+struct fpu_reset_guard{
```

Comment:
I like what you did here with turning the call into a class where the destructor is fired at the end.